src/libraries/tick_bitmap.cairo

```diff
@@ -205,4 +124,60 @@ mod TickBitmap {
             }
         }
     }
+
+    fn generate_hashed_word_pos(word_pos: @i16) -> felt252 {
+        let mut serialized: Array<felt252> = ArrayTrait::new();
+        Serde::<i16>::serialize(word_pos, ref serialized);
+        poseidon_hash_span(serialized.span())
+    }
+
+    /// Calculates the word value based on a tick input.
+    /// - For ticks between 0 and 255 inclusive, it returns 0.
+    /// - For ticks greater than 255, it divides the tick by 256.
+    /// - For ticks less than 0 but greater than or equal to -256, it returns -1.
+    /// - For other negative ticks, it divides the tick by 256 and subtracts 1.
+    ///
+    /// Parameters:
+    /// - `tick`: An i32 input representing the tick value.
+    ///
+    /// Returns: An i16 value representing the calculated word.
+    fn calculate_word(tick: i32) -> i16 {
+        let zero = IntegerTrait::<i32>::new(0, false);
+        let one_negative = IntegerTrait::<i32>::new(1, true);
+        let upper_bound = IntegerTrait::<i32>::new(255, false);
+        let divisor = IntegerTrait::<i32>::new(256, false);
+        let negative_lower_bound = IntegerTrait::<i32>::new(256, true);
+
+        let result = if tick >= zero && tick <= upper_bound {
+            zero
+        } else if tick > upper_bound {
+            tick / divisor
+        } else if tick >= negative_lower_bound {
+            one_negative
+        } else {
+            tick / divisor + one_negative
+        };
+        result.try_into().expect('calculate_word')
```

Comment:
We should discuss for a future refactor to remove all the `expects` for functions that are not internal or external. This functions should return a Result or Error and then the internal or external function should decide if it panics. The panic logic should be managed by the "Smart Contract" functions. What do you think?

Reply:
@dubzn what do you think?

Reply:
I agree, if it's something we can take to a library, we could even make these functions return an `Option<i32>` and decide what to do in higher layers.